crates/oxc_linter/src/config/settings/jsdoc.rs

```diff
@@ -4,79 +4,114 @@ use serde::Deserialize;
+    /// For all rules but NOT apply to `check-access` and `empty-tags` rule
     #[serde(default, rename = "ignorePrivate")]
     pub ignore_private: bool,
-    /// Not apply to `empty-tags` rule
+    /// For all rules but NOT apply to `empty-tags` rule
     #[serde(default, rename = "ignoreInternal")]
     pub ignore_internal: bool,
-    /// Apply to `require-*` rules only
+
+    /// Only for `require-(yields|returns|description|example|param|throws)` rule
     #[serde(default = "default_true", rename = "ignoreReplacesDocs")]
     pub ignore_replaces_docs: bool,
-    /// Apply to `require-*` rules only
+    /// Only for `require-(yields|returns|description|example|param|throws)` rule
     #[serde(default = "default_true", rename = "overrideReplacesDocs")]
     pub override_replaces_docs: bool,
-    /// Apply to `require-*` rules only
+    /// Only for `require-(yields|returns|description|example|param|throws)` rule
     #[serde(default, rename = "augmentsExtendsReplacesDocs")]
     pub arguments_extends_replaces_docs: bool,
-    /// Apply to `require-*` rules only
+    /// Only for `require-(yields|returns|description|example|param|throws)` rule
     #[serde(default, rename = "implementsReplacesDocs")]
     pub implements_replaces_docs: bool,
 
+    /// Only for `require-param-type` and `require-param-description` rule
+    #[serde(default, rename = "exemptDestructuredRootsFromChecks")]
+    pub exempt_destructured_roots_from_checks: bool,
+
     #[serde(default, rename = "tagNamePreference")]
     tag_name_preference: FxHashMap<String, TagNamePreference>,
-
+    //
     // Not planning to support?
-    // min_lines
-    // max_lines
-
-    // XXX: Need more investigation to understand these...
-    // mode: String,
-    // preferred_types: FxHashMap<String, PreferredType>,
-    // structured_tags: FxHashMap<String, StructuredTag>,
-    // contexts: Vec<Context>,
+    // min_lines: number
+    // max_lines: number
+    // mode: string
+    //
+    // TODO: Need more investigation to understand these usage...
+    //
+    // Only for `check-types` and `no-undefined-types` rule
+    // preferred_types: Record<
+    //   string,
+    //   false | string | {
+    //     message: string;
+    //     replacement?: false | string;
+    //     skipRootChecking?: boolean;
+    //   }
+    // >
+    //
+    // structured_tags: Record<
+    //   string,
+    //   {
+    //     name?: "text" | "namepath-defining" | "namepath-referencing" | false;
+    //     type?: boolean | string[];
+    //     required?: ("name" | "type" | "typeOrNameRequired")[];
+    //   }
+    // >
+    //
+    // I know this but not sure how to implement
+    // contexts: string[] | {
+    //   disallowName?: string;
+    //   allowName?: string;
+    //   context?: string;
+    //   comment?: string;
+    //   tags?: string[];
+    //   replacement?: string;
+    //   minimum?: number;
+    //   message?: string;
+    //   forceRequireReturn?: boolean;
+    // }[]
```

Comment:
I think the rule of thumb is to add more data when required, otherwise we may end up implementing useless stuff.

Reply:
Yes, I agree! 
(But the most difficult part of this plugin is that everything is tightly coupled and completely unpredictable where any of them will be needed. 😂)